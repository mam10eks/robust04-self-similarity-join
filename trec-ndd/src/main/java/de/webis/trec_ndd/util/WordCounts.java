package de.webis.trec_ndd.util;

import java.io.IOException;
import java.time.temporal.ChronoUnit;

import org.netspeak.client.Netspeak;
import org.netspeak.client.Request;

import com.google.common.cache.CacheBuilder;
import com.google.common.cache.CacheLoader;
import com.google.common.cache.LoadingCache;

import lombok.SneakyThrows;
import lombok.experimental.UtilityClass;
import net.jodah.failsafe.Failsafe;
import net.jodah.failsafe.RetryPolicy;

@UtilityClass
public class WordCounts {
	private static final String NETSPEAK_API = "https://api.netspeak.org/netspeak3/search?";
	
	public static RetryPolicy<Long> retryPolicy = new RetryPolicy<Long>()
			  .handle(Exception.class)
			  .withBackoff(5, 180, ChronoUnit.SECONDS)
			  .withMaxRetries(3);
	
	private static final LoadingCache<String, Long> WORD_COUNT = CacheBuilder.newBuilder()
			.build(CacheLoader.from(WordCounts::wordCountFailsave));

	@SneakyThrows
	public static Long getWordCount(String word) {
		return WORD_COUNT.get(word);
	}
	
	private static Long wordCountFailsave(String word) {
		return Failsafe.with(retryPolicy)
				.get(() -> getWordCountOrFail(word));
	}
	
	private static long getWordCountOrFail(String word) throws IOException {
		Request request = new Request();
		request.put(Request.QUERY, word);
		
		return new Netspeak(NETSPEAK_API)
				.search(request).getTotalFrequency();
	}
}
