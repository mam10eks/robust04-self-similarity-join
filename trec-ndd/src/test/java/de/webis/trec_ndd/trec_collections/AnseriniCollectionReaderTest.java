package de.webis.trec_ndd.trec_collections;

import java.util.List;

import org.approvaltests.Approvals;
import org.junit.Test;

import io.anserini.collection.SourceDocument;

public class AnseriniCollectionReaderTest<T extends SourceDocument> {
	@Test
	public void approveTransformationOfSmallCollection() {
		CollectionReader<T> reader = robustAnseriniCollectionReader();
		List<CollectionDocument> documents = reader.extractJudgedDocumentsFromCollection();

		Approvals.verifyAsJson(documents);
	}
	
	@Test
	public void approveTransformationOfSmallCluewebSampleCollection() {
		CollectionReader<T> reader = cluewebAnseriniCollectionReader();
		List<CollectionDocument> documents = reader.extractJudgedDocumentsFromCollection();

		Approvals.verifyAsJson(documents);
	}

	private static <T extends SourceDocument> CollectionReader<T> robustAnseriniCollectionReader() {
		String pathToCollection = "../collection_to_doc_vectors/test/data/robust";
		String pathToQrels = "../collection_to_doc_vectors/test/data/robust-qrels.txt";
		String collectionType = "TrecCollection";
		String documentGenerator = "JsoupGenerator";

		return new AnseriniCollectionReader<T>(pathToCollection, pathToQrels, collectionType, documentGenerator);
	}
	
	private static <T extends SourceDocument> CollectionReader<T> cluewebAnseriniCollectionReader() {
		String pathToCollection = "../collection_to_doc_vectors/test/data/clueweb09-sample";
		String pathToQrels = "../collection_to_doc_vectors/test/data/clueweb09-qrels.txt";
		String collectionType = "ClueWeb09Collection";
		String documentGenerator = "JsoupGenerator";

		return new AnseriniCollectionReader<T>(pathToCollection, pathToQrels, collectionType, documentGenerator);
	}
}
