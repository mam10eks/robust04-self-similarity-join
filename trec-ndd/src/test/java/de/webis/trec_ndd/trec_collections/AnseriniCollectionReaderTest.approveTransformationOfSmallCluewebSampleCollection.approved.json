[
  {
    "id": "clueweb09-en0000-00-00000",
    "content": "00000-NRT-RealEstate\u0027s Homepage Startup go home 00000-NRT-RealEstate\u0027s NRT LLC Burgdorf.com 3 No Website Set Coldwell Banker Burnet CBDFW.com CBMove.com 7 No Website Set CBSuccess.com CBWS.com 10 No Website Set 11 No Website Set 12 No Website Set Welcome to 00000-NRT-RealEstate\u0027s Homepage What is this? | Create Your Own Homepage  Change My Search To: Google Search MSN Search Yahoo! Search Ask.com Search Wikipedia English Yahoo! Answers Answers.com YouTube Videos"
  },
  {
    "id": "clueweb09-en0000-00-00010",
    "content": "Contact - 00perdomain.com  Home              Contact     Phone     1 - 8 8 8 - 2 - z a z C O R P 1 - 8 8 8 - 2 9 2 - 9 2 6 7 San Francisco  New York City Denver  Seattle  Phoenix  Los Angeles Boston  Chicago  Washington, DC International (outside the U.S.) E-mail     General Inquiry Sales Department Employment / HR Fax     1 - 9 2 5 - 9 0 5 - 0 4 5 7       E-Mail Password             Home | Contact | Policies Copyright © 1995-2008, zaz Corporation. All rights reserved. Privacy Policy"
  },
  {
    "id": "clueweb09-en0000-00-00021",
    "content": "Vaktydskrifte - Nuus, Afrikaans - 00perdomain.com  Nuus,  Afrikaans,  World  –  Top      Vaktydskrifte        LexikosLexikos Leksikografie, uitgegee deur die WAT by die Universiteit Stellenbosch. Argief van inhoudsopgawes en uitgesoekte artikels. www.sun.ac.za     E-Mail Password       E X P A N D      m i n i m i z e             Home | Contact | Policies  Help build the largest human-edited directory on the web.  Submit a Site - Open Directory Project - Become an Editor Copyright ©1995-2008, zaz Corporation. All rights reserved. Privacy Policy"
  }
]